datacite-doi-commons/src/main/java/no/unit/nva/datacite/DynamoRecordDtoMapper.java

```diff
@@ -0,0 +1,74 @@
+package no.unit.nva.datacite;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+import no.unit.nva.publication.doi.dto.Contributor;
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.PublicationType;
+import no.unit.nva.transformer.dto.CreatorDto;
+import no.unit.nva.transformer.dto.DynamoRecordDto;
+import no.unit.nva.transformer.dto.IdentifierDto;
+import no.unit.nva.transformer.dto.PublisherDto;
+import no.unit.nva.transformer.dto.ResourceTypeDto;
+import no.unit.nva.transformer.dto.TitleDto;
+
+public final class DynamoRecordDtoMapper {
+
+    private DynamoRecordDtoMapper() {
+    }
+
+    /**
+     * Maps a Publication to DynamoRecordDto. For use in the nva doi partner data Transformer.
+     *
+     * @param publication   publication
+     * @return  dynamoRecordDto
+     */
+    public static DynamoRecordDto fromPublication(Publication publication) {
+        return new DynamoRecordDto.Builder()
+            .withCreator(toCreatorDtoList(publication.getContributor()))
```

Comment:
are all these fields mandatory and guaranteed never to be null?

Reply:
No, good input